### PR TITLE
Downgrade @hot-loader/react-dom and fix make dev

### PR DESCRIPTION
**Pull request by hmhealey**
When doing the dependency upgrades, I didn't know how to use `make dev`, so I didn't realize it was broken. This fixes it by:
1. Downgrading `@hot-loader/react-dom` since I accidentally upgraded it when we weren't ready to upgrade to React 17 yet.
2. Switching to the newer way of starting webpack-dev-server using `webpack serve`
3. Fixing the name of the source map type since the old name must've been removed in Webpack 5.

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/7512

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -242,7 +242,7 @@
     "fix": "eslint --ext .js,.jsx,.tsx,.ts . --quiet --fix --cache",
     "build": "cross-env NODE_ENV=production webpack",
     "run": "webpack --progress --watch",
-    "dev-server": "webpack-dev-server -d",
+    "dev-server": "webpack serve",
```
Comment (ethervoid) [inline package.json:245]: Probably we want to pass `--env development --mode development`?

**Review by ethervoid — APPROVED** [review]
LGTM! Just a minor comment

File: package.json (hunk 1 above)
Comment (hmhealey) [reply package.json:245]: Good call. I'll add that

**Review by hahmadia — APPROVED** [review]

**Review by AGMETEOR — APPROVED** [review]

**Pull request merged**